### ISSUE TITLE
Node save with file fields =  multiple nodes

### ISSUE THREAD

**Issue by chrisschaub** [signalpoint/DrupalGap#400]
So, we have single occurring image fields on a node. When we save the node via the app, we get a node saved for each image. These image fields only support single uploads. So, if you have 4 image fields with four images, each successive node that gets created has an additional image. The last one has all 4 saved. This wasn't happening and started recently. Trying to track it down but thought I'd write in case anything came to mind. Thanks.

**Comment by signalpoint** [comment]
@chrisschaub Here are the logs: https://github.com/signalpoint/DrupalGap/commits/7.x-1.x/src/modules/image

Are you saying that one node form has 4 individual image fields, and when you upload one node with 4 images, 4 nodes get created?

**Comment by chrisschaub** [comment]
Yes

**Comment by chrisschaub** [comment]
I have tested this with a clean drupal and drupalgap, bug confirmed. This wasn't happening and just started recently. I wonder if something in entity_create is forcing the node create each time it's called? I don't see anything glaring on the drupalgap side codewise.

**Comment by signalpoint** [comment]
@chrisschaub If you set `Drupal.settings.debug` to `true`, does it make 4 POST/PUT calls to the server? Please share the console log info here, I'm curious.

**Comment by chrisschaub** [comment]
Yeah, its calling 

POST: http://drupal.dev/?q=drupalgap/node.json 

once for each file.

**Comment by signalpoint** [comment]
@chrisschaub The problem will lie somewhere in _drupalgap_form_submit() and/or _image_field_form_process(). I believe one or both of those functions needs to be expanded to understand multiple image fields.

As it stands right now, soon each processed image will continue onward with the form submission, when instead only the last image field should continue onward with the form validation/submission.

As you can see here (https://github.com/signalpoint/DrupalGap/blob/7.x-1.x/src/includes/form.inc.js#L1176), the form validation will be called every time there is an image field, which in turn will call the submit function, hence multiple submissions.

I believe this has always been an issue, and it was merely covered up by only processing 1 image field. Which stopped happening ever since we removed the single boolean line (`return false;` - https://github.com/signalpoint/DrupalGap/commit/acdac43fce77c8ef4f4ccb53a46b8a0f05881855) of code which prevented multiple image fields from being added to a form. That boolean was probably there because of this exact situation (but I can't remember).

TLDR, the image processing calls need to be chained, so once one image is processed, it then processes the next, and so on. The last image to be processed would then continue on by calling the options.success() (aka form validation).
